server/src/main/java/org/elasticsearch/snapshots/RestoreService.java

```diff
@@ -776,51 +774,67 @@ private IndexMetadata updateIndexSettings(IndexMetadata indexMetadata,
             return builder.settings(settingsBuilder).build();
         }
 
-        private void restoreTemplates(Map<RelationName, String> templates,
+        private boolean shouldRename() {
+            return !request.tableRenamePattern().equals(TABLE_RENAME_PATTERN.getDefault(Settings.EMPTY))
+                || !request.tableRenameReplacement().equals(TABLE_RENAME_REPLACEMENT.getDefault(Settings.EMPTY))
+                || !request.schemaRenamePattern().equals(SCHEMA_RENAME_PATTERN.getDefault(Settings.EMPTY))
+                || !request.schemaRenameReplacement().equals(SCHEMA_RENAME_REPLACEMENT.getDefault(Settings.EMPTY));
+        }
+
```

Comment:
Let's use it for `applyRenameToIndices` as well

Reply:
Good call. I created a property on the request class